example_client/benchmark/metrics.py

```diff
@@ -0,0 +1,352 @@
+            self[f"{prefix}cv_latency"] = self[f"{prefix}stdev_latency"] / mean_lat
+        except:
+            self[f"{prefix}stdev_latency"] = 0.0
+            self[f"{prefix}cv_latency"] = 0.0
+        self[f"{prefix}mean_latency2"] = mean_lat2
+        self[f"{prefix}mean_latency"] = mean_lat
+
+        if f"{prefix}pass_mean_latency" not in self: self[f"{prefix}pass_mean_latency"] = 0.0
+        if f"{prefix}pass_mean_latency2" not in self: self[f"{prefix}pass_mean_latency2"] = 0.0
+        if f"{prefix}pass_mean_latency" not in other: other[f"{prefix}pass_mean_latency"] = 0.0
+        if f"{prefix}pass_mean_latency2" not in other: other[f"{prefix}pass_mean_latency2"] = 0.0
+        mean_lat2 = ((N - 1) * self[f"{prefix}pass_mean_latency2"] + other[f"{prefix}pass_mean_latency2"]) / N
+        mean_lat = ((N - 1) * self[f"{prefix}pass_mean_latency"] + other[f"{prefix}pass_mean_latency"]) / N
+        try:
+            self[f"{prefix}pass_stdev_latency"] = math.sqrt(mean_lat2 - mean_lat ** 2)
+            self[f"{prefix}pass_cv_latency"] = self[f"{prefix}pass_stdev_latency"] / mean_lat
+        except:
+            self[f"{prefix}pass_stdev_latency"] = 0.0
+            self[f"{prefix}pass_cv_latency"] = 0.0
+        self[f"{prefix}pass_mean_latency2"] = mean_lat2
+        self[f"{prefix}pass_mean_latency"] = mean_lat
+
+        if f"{prefix}fail_mean_latency" not in self: self[f"{prefix}fail_mean_latency"] = 0.0
+        if f"{prefix}fail_mean_latency2" not in self: self[f"{prefix}fail_mean_latency2"] = 0.0
+        if f"{prefix}fail_mean_latency" not in other: other[f"{prefix}fail_mean_latency"] = 0.0
+        if f"{prefix}fail_mean_latency2" not in other: other[f"{prefix}fail_mean_latency2"] = 0.0
+        mean_lat2 = ((N - 1) * self[f"{prefix}fail_mean_latency2"] + other[f"{prefix}fail_mean_latency2"]) / N
+        mean_lat = ((N - 1) * self[f"{prefix}fail_mean_latency"] + other[f"{prefix}fail_mean_latency"]) / N
+        try:
+            self[f"{prefix}fail_stdev_latency"] = math.sqrt(mean_lat2 - mean_lat ** 2)
+            self[f"{prefix}fail_cv_latency"] = self[f"{prefix}pass_stdev_latency"] / mean_lat
+        except:
+            self[f"{prefix}fail_stdev_latency"] = 0.0
+            self[f"{prefix}fail_cv_latency"] = 0.0
+        self[f"{prefix}fail_mean_latency2"] = mean_lat2
+        self[f"{prefix}fail_mean_latency"] = mean_lat
+
+        for key in other.keys():
+            if re.search(f"^{prefix}hist_latency_", key):
+                try: self[key] += other[key]
+                except KeyError: self[key] = other[key]
+
+    def __iadd__(self, other):
+        self.calc_sum(other, "submetrics")
+        self.__iadd_series("warmup_", other)
+        self.__iadd_series("window_", other)
+        self.__iadd_series("", other)
+        return self
+
+
+    def calc_quantile_value(self, latlist, total, quantile):
+        assert quantile >= 0 and quantile <= 1
+        if (1 - quantile) * total < 6:
```

Comment:
Why 6? Can we name this value so the name reflects the purpose?

Reply:
added: minimal_samples_number = 6